sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -61,10 +61,213 @@ This sub-client has been implemented. Refer to [ModulesClient](./ModulesClient.c
+    /// <param name="excludeKeys">If false, authorization keys are included in export output.</param>
+    /// <param name="options">The optional settings for this request.</param>
+    /// <param name="cancellationToken">Task cancellation token.</param>
+    /// <returns>JobProperties of the newly created job.</returns>
+    public virtual Task<Response<JobProperties>> CreateExportDevicesJobAsync(string outputBlobContainerUri, bool excludeKeys, JobRequestOptions options = default, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Creates a job to import device registrations into the IoT Hub.
+    /// </summary>
+    /// <param name="importBlobContainerUri">URI containing SAS token to a blob container that contains registry data to sync.</param>
+    /// <param name="outputBlobContainerUri">URI containing SAS token to a blob container. This is used to output the status of the job.</param>
+    /// <param name="options">The optional settings for this request.</param>
+    /// <param name="cancellationToken">Task cancellation token.</param>
+    /// <returns>JobProperties of the newly created job.</returns>
+    public virtual Task<Response<JobProperties>> CreateImportDevicesJobAsync(string importBlobContainerUri, string outputBlobContainerUri, JobRequestOptions options = default, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// List all import and export jobs for the IoT Hub.
+    /// </summary>
+    /// <param name="cancellationToken">Task cancellation token</param>
+    /// <returns>IEnumerable of JobProperties of all jobs for this IoT Hub.</returns>
+    public virtual Task<Response<IEnumerable<JobProperties>>> GetImportExportJobsAsync(CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Gets the import or export job with the specified ID.
+    /// </summary>
+    /// <param name="jobId">Id of the Job object to retrieve</param>
+    /// <param name="cancellationToken">Task cancellation token</param>
+    /// <returns>JobProperties of the job specified by the provided jobId.</returns>
+    public virtual Task<Response<JobProperties>> GetImportExportJobAsync(string jobId, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Cancels/Deletes the job with the specified ID.
+    /// </summary>
+    /// <param name="jobId">Id of the job to cancel</param>
+    /// <param name="cancellationToken">Task cancellation token</param>
+    /// <returns>A response string object indicating result of the cancellation.</returns>
+    public virtual Task<Response<string>> CancelImportExportJobAsync(string jobId, CancellationToken cancellationToken = default);
+}
+
+```
+
+## Scheduled Jobs
+Scheduled jobs execute device twin updates and direct methods against a set of devices at a scheduled time. You can use scheduled jobs to update desired properties, update tags and invoke direct methods.
+
+```csharp
+public class ScheduledJobs
```

Comment:
Do we need a separate class for Scheduled Jobs?

Reply:
When we discussed this during our last design discussion meeting, it seems like folks prefer having a separate class to reduce confusion on the user. There are two types of jobs, import/export jobs (jobs/v1), and scheduled jobs(jobs/v2). Having both sets in a single class can add confusion in a way where users might think import/export jobs could also be scheduled jobs. Having them separated adds clarity and reduces confusion to the user, IMO. But it's still open for discussion.

Reply:
Will combine both classes per your recommendation during the design meeting.